util/util.go

```diff
@@ -451,9 +453,27 @@ func IsLiteral(expr ast.Expr, literals ...string) bool {
 	return false
 }
 
+// ExprToString converts AST expression to string
+func ExprToString(e ast.Expr, pass *analysis.Pass) string {
+	var buf bytes.Buffer
+	err := printer.Fprint(&buf, pass.Fset, e)
+	if err != nil {
+		panic(fmt.Sprintf("Failed to convert AST expression to string: %v\n", err))
+	}
+	return buf.String()
+}
+
+// TruncatePosition truncates the prefix of the filename to keep it at the given depth (config.DirLevelsToPrintForTriggers)
+func TruncatePosition(position token.Position) token.Position {
+	position.Filename = PortionAfterSep(
+		position.Filename, "/",
+		config.DirLevelsToPrintForTriggers)
+	return position
+}
+
 var codeReferencePattern = regexp.MustCompile("\\`(.*?)\\`")
 var pathPattern = regexp.MustCompile(`"(.*?)"`)
-var nilabilityPattern = regexp.MustCompile(`([\(|^\t](?i)(definitely\s|must\sbe\s)(nilable|nonnil)[\)]?)`)
+var nilabilityPattern = regexp.MustCompile(`([\(|^\t](?i)(found\s|must\sbe\s)(nilable|nonnil)[\)]?)`)
 
 // PrettyPrintErrorMessage is used in error reporting to post process and pretty print the output with colors
 func PrettyPrintErrorMessage(msg string) string {
```

Comment:
Is it possible that we integrate (in the future of course) this pretty print logic into the diagnostics engine organically? 

Post-process via regex always feels a bit hacky to me :)

Reply:
Agreed, should be doable. I'll try and incorporate this in the planned refactor.